extensions/sql-migration/src/wizard/skuRecommendationPage.ts

```diff
@@ -709,7 +722,7 @@ export class SKURecommendationPage extends MigrationWizardPage {
 
 						if (this.hasRecommendations()) {
 							const successfulRecommendationsCount =
-								this.migrationStateModel._skuRecommendationResults.recommendations.sqlDbRecommendationResults
+								this.migrationStateModel._skuRecommendationResults.recommendations!.sqlDbRecommendationResults
 									.filter(r => r.targetSku !== null).length;
 							this._rbg.cards[index].descriptions[CardDescriptionIndex.SKU_RECOMMENDATION].textValue =
 								constants.RECOMMENDATIONS_AVAILABLE(successfulRecommendationsCount);
```

Comment:
so looks like the logic here is different than for the other skus. just curious what's different about this case? 

I see that we're not setting "recommendation " here, but i'm a bit confused. Shouldn't we be setting ServiceTier here like we do for MI scenario? Also why don't we have to take elastic vs baseline model into account here?

Reply:
yeah, good question. the difference is that MI and VM are guaranteed to only have one recommendation result (since the recommendation is for the full instance), so we can just show a small preview of the recommendation in the MI/VM results. but for SQL DB, we get multiple recommendation results, one per database. so there isn't really any preview we can show that would make sense, so we just show the count of recommendations instead -- then the results can be seen when you view the recommendation details

Reply:
so to answer your question, there is no one single `recommendation` object because there isn't just one in this case -- although you do bring up a good point about taking into account elastic vs baseline -- the `successfulRecommendationsCount` should count how many recommendations were returned by the model that the user asked for, currently it always defaults to the baseline model. nice catch!